java/arcs/core/util/MutableBiMap.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.util
+
+/**
+ * Mutable bi-directional map utility class.
+ */
+class MutableBiMap<L, R>() {
+    private val left2right: MutableMap<L, R> = mutableMapOf()
+    private val right2left: MutableMap<R, L> = mutableMapOf()
+
+    val size: Int get() = left2right.size
+    val entries: MutableSet<MutableMap.MutableEntry<L, R>> get() = left2right.entries
+    val lefts: MutableSet<L> get() = left2right.keys
+    val rights: MutableSet<R> get() = right2left.keys
```

Comment:
These should probably be private.

Reply:
`entries`, `keys`, and `values` are public for both `Map` and `MutableMap` implementations, so I think it makes sense for `entries`, `lefts`, and `rights` to be public.